### ISSUE TITLE
Underline effect not working in tmux.

### ISSUE THREAD

**Issue by Addisonbean** [gyscos/Cursive#222]
When using tmux, Effect::Underline has no effect. I confirmed that other programs are able to output underlined text, but for some reason it's not working when in tmux.

Here's a simplified example of what I'm using that still recreates the problem:

```
extern crate cursive;

use cursive::Cursive;
use cursive::views::{Dialog, TextView};
use cursive::theme::Effect;

fn main() {
    let mut siv = Cursive::new();

    siv.add_layer(Dialog::around(TextView::new("Hello Dialog!").effect(Effect::Underline))
                         .title("Cursive")
                         .button("Quit", |s| s.quit()));

    siv.run();
}

```

# Environment

* Operating system: macOS 10.12.6
* Tmux version: 2.5
* Backend used: ncurses
* Current locale:

```
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL="en_US.UTF-8"
```

* Cursive version: 0.8.1

**Comment by gyscos** [comment]
Hi, and thanks for the report!

I just tried it locally (though on Linux, not macOS), and I do get underline text, both in and out of tmux, and with any of the `ncurses`, `pancurses` or `termion` backends.

So we'll need a few more information to narrow down the problem:
* Does it work when outside of tmux?
* Does it work with a different backend? (See [the wiki page](https://github.com/gyscos/Cursive/wiki/Backends))
* What does `echo $TERM` return when in tmux?
* Do you have a `~/.tmux.conf` file? If so, what is the content? Can you rename that file (for instance to `~/.tmux.conf.backup`), start a new tmux session, and compare the results?

**Comment by Addisonbean** [comment]
Here's the value of `$TERM`:
`screen-256color`

I checked everything in my `~/.tmux.conf`, and found out that it was this line that caused the issue:

`set -g default-terminal "screen-256color"`

However without that, none of the custom cursive colors I'm using work at all. They all just appear black.

**Comment by Addisonbean** [comment]
I just tried the other backends, and it worked inside tmux with the `termion` backend, but none of the others. They all worked outside of tmux.

Oh and here's my ncurses version:
```
#define NCURSES_VERSION_MAJOR 5
#define NCURSES_VERSION_MINOR 7
#define NCURSES_VERSION_PATCH 20081102
```

**Comment by gyscos** [comment]
Ah indeed, it seems the default `TERM` value for tmux is `screen`, which causes issues with the colors re-definition. That's interesting, and should probably be an issue by itself.

As for our problem, let's test ncurses itself.

Write this to `test.c`:

```c
#include <ncurses.h>

int main(int argc, char* argv[])
{
    initscr();
    attron(A_UNDERLINE);
    printw("Press a key to continue");
    attroff(A_UNDERLINE);

    getch();

    endwin();
    return 0;
}
```

And run `g++ test.c -lncurses && ./a.out` from the tmux session.

If the text is not underlined, it is a ncurses issue.

**Comment by Addisonbean** [comment]
I run that program and the text was underlined. But updating ncurses did fix it! I am on 6.1 now, and it's working with the default backend in tmux now.

**Comment by gyscos** [comment]
Yay! Some constants defined in `ncurses.h` are hard-coded in `ncurses-rs`, and I suppose the value for `A_UNDERLINE` may have changed between versions, leading to some confusion when they mismatched.

Anyway, onto the colors issue now!
